UNITTESTS/CMakeLists.txt

```diff
@@ -1,240 +1,68 @@
-cmake_minimum_required(VERSION 3.0.2)
-
-set(PROJECT_NAME unittests)
-set(LIB_NAME MbedOS)
-
-project(${PROJECT_NAME})
-
-# Setup c++ standard
-macro(use_cxx14)
-    if (CMAKE_VERSION VERSION_LESS 3.1)
-        if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14")
-        endif()
-    else()
-        set(CMAKE_CXX_STANDARD 14)
-        set(CMAKE_CXX_STANDARD_REQUIRED ON)
-    endif()
-endmacro()
-
-use_cxx14()
-
-
-if (MINGW)
-  # enable PRIx formatting globally
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D__STDC_FORMAT_MACROS")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STDC_FORMAT_MACROS")
-endif (MINGW)
-
-####################
-# GTEST
-####################
-
-# Download and unpack googletest at configure time
-configure_file(googletest-CMakeLists.txt.in googletest-download/CMakeLists.txt)
-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-    RESULT_VARIABLE result
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/googletest-download)
-if (result)
-    message(FATAL_ERROR "CMake failed for google test: ${result}")
-endif()
-execute_process(COMMAND ${CMAKE_COMMAND} --build .
-    RESULT_VARIABLE result
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/googletest-download)
-if (result)
-    message(FATAL_ERROR "Build failed for google test: ${result}")
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include(FetchContent)
+
+# Download and unpack googletest
+if(NOT GTEST_FOUND)
```

Comment:
I have wrongly mixed this variable with `fetchcontent_xyz` calls as this suppose to be used with the find_pacakage command so I will remove this condition check.
`FetchContent_MakeAvailable` is already taken care of to avoid the second time download when we called the same command again